neps/nep-0452.md

```diff
@@ -0,0 +1,234 @@
+
+- calls `claim`. The transaction is signed on behalf of the linkdrop contract (`signer_id` is set to the contract address) using `privKey1`.
+  - the args of this function call will simply contain the user's existing account ID.
+  - the linkdrop contract will delete the access key associated with `pubKey1` so that it cannot be used again.
+  - the linkdrop contract will transfer the funds to that account alongside any other assets.
+
+```ts
+/// Information about a specific public key.
+type KeyInfo = {
+   /// How much Gas should be attached when the key is used to call `claim` or `create_account_and_claim`.
+   /// It is up to the smart contract developer to calculate the required gas (which can be done either automatically on the contract or on the client-side).
+   required_gas: string,
+
+   /// yoctoNEAR$ amount that will be sent to the account that claims the linkdrop (either new or existing)
+   /// when the key is successfully used.
+   yoctoNEAR: string,
+
+   /// If using the NFT standard extension, a set of NFTData can be linked to the public key
+   /// indicating that all those assets will be sent to the account that claims the linkdrop (either new or
+   /// existing) when the key is successfully used.
+   nft_data: NFTData[] | null,
+
+   /// If using the FT standard extension, a set of FTData can be linked to the public key
+   /// indicating that all those assets will be sent to the account that claims the linkdrop (either new or
+   /// existing) when the key is successfully used.
+   ft_data: FTData[] | null
+
+   /// ... other types can be introduced and the standard is easily extendable.
+}
+
+
+/// Data outlining a specific Non-Fungible Token that should be sent to the claiming account
+/// (either new or existing) when a key is successfully used.
+type NFTData = {
+   /// the id of the token to transfer
+   token_id: string,
+
+   /// The valid NEAR account indicating the Non-Fungible Token contract.
+   contract_id: string
+}
+
+
+/// Data outlining Fungible Tokens that should be sent to the claiming account
+/// (either new or existing) when a key is successfully used.
+type FTData = {
+   /// The number of tokens to transfer, wrapped in quotes and treated
+   /// like a string, although the number will be stored as an unsigned integer
+   /// with 128 bits.
+   amount: string,
+
+   /// The valid NEAR account indicating the Fungible Token contract.
+   contract_id: string
+}
+
+
+
+
+/****************/
+/* VIEW METHODS */
+/****************/
+
+/// Returns the $yoctoNEAR amount associated with a given public key
+/// Panics if the key does not exist.
+/// The Public key must be in a binary format with base58 string serialization with human-readable curve.
+/// The key types currently supported are secp256k1 and ed25519. Ed25519 public keys accepted are 32 bytes
+/// and secp256k1 keys are the uncompressed 64 format.
```

Comment:
Please document how the private key is serialized to the string.

Reply:
I will add a comment forcing serialization to be done with borsh. If dApps choose to deviate from the standard serialization, they should be considered non standard. Handling different serialization types for the private key is out of scope for this NEP.

Reply:
Can we serialize not as Borsh? It's a custom serialization. Since this is a return of a public function, we should use something standard. My proposals:
* base58 (preferred)
* base64
* base32

Reply:
@BenKurrek This description is misleading, indeed. I hope you did not intend to reinvent the current state of things, which is: [key (binary array) is serialized with base58, and prefixed the key type](https://github.com/near/nearcore/blob/master/core/crypto/src/signature.rs#L184), e.g. `ed25519:9VNCBovVBqSjP375PzJgn4J1LofeLkxEeq8pEvQkJd5Z` (borsh is not involved here). Here is an example from a different NEP:

https://github.com/near/NEPs/blob/3657715aa30ca472bdc78399e4cfd3ec3805bf86/neps/nep-0413.md?plain=1#L142

Reply:
Oops, yea that was not my intention. I will copy this comment.